tests/state_test_util.go

```diff
@@ -240,6 +240,9 @@ func (t *StateTest) RunNoVerify(subtest StateSubtest, vmconfig vm.Config, snapsh
 	vmconfig.ExtraEips = eips
 
 	block := t.genesis(config).ToBlock()
+	// MakePreState invokes snapshot.New, which will start a goroutine to
+	// generate.go:generate(). This goroutine cannot exit until the abort-stats
+	//  can be delivered, either via a call to Journal or e.g Disable.
 	triedb, snaps, statedb := MakePreState(rawdb.NewMemoryDatabase(), t.json.Pre, snapshotter, scheme)
```

Comment:
How about adding this here?

```
if snaps != nil {
    defer snaps.Disable()
}
```

Reply:
And while here, we should probably also use `defer triedb.Close()`.

Reply:
We should only close these things _unless_ we return them to caller. If we do, it's the callers job to clean up

Reply:
Ahh, but that's a design smell then. So there are quite a few cases where we need to take care of closing it. The other way of using `defer` here would be naming the returns, and then adding:

```
defer func() {
    if err != nil {
       triedb.Close()
       if snaps != nil {
           snaps.Disable()
       }
    }
}()
```

Reply:
Yeah, I agree. Once the CI runs is done, I can change it to use that defer

Reply:
This made the tests fail, for some reason https://ci.appveyor.com/project/ethereum/go-ethereum/builds/49167581/job/flhcsji6lgpk99kt#L1068